internal/cmd/context/auth.go

```diff
@@ -51,28 +54,40 @@ and then attach it to one or more contexts.`,
 
 	// add cmd flags
 	flags := cmd.Flags()
-	options.AddBasicAuthFlags(flags)
+	options.AddContextcAuthFlags(flags)
 
 	// add sub commands
 
 	return cmd
 }
 
 func setAuth(authName string, opts *clioptions.CLIOptions) (bool, error) {
+	if len(opts.JWTJsonPath) > 0 && (len(opts.BasicClientID) > 0 || len(opts.BasicClientSecret) > 0) {
+		return false, fmt.Errorf("is not possible to set bot jwt and basic service account configs")
+	}
+
 	locator := cliconfig.NewConfigPathLocator()
 	locator.ExplicitPath = opts.MiactlConfig
+	if len(opts.JWTJsonPath) > 0 {
+		return saveJWTServiceAccount(authName, opts.JWTJsonPath, locator)
+	}
+	return saveBasicServiceAccount(authName, opts.BasicClientID, opts.BasicClientSecret, locator)
+}
 
+func saveBasicServiceAccount(name, clientID, clientSecret string, locator *cliconfig.ConfigPathLocator) (bool, error) {
 	config, err := locator.ReadConfig()
 	if err != nil {
 		return false, err
 	}
 
 	newAuth := &api.AuthConfig{
-		ClientID:     opts.BasicClientID,
-		ClientSecret: opts.BasicClientSecret,
+		ClientID:          clientID,
+		ClientSecret:      clientSecret,
+		JWTKeyID:          "",
+		JWTPrivateKeyData: "",
```

Comment:
With set this as empty string? It is the default value, you could avoid it

Reply:
The object are merged with possible values already present on the context. I'm setting them explicitly for override the values